docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md

```diff
@@ -0,0 +1,599 @@
+                      },
+                      {
+                        property: "og:image:width",
+                        content: metaImage.width,
+                      },
+                      {
+                        property: "og:image:height",
+                        content: metaImage.height,
+                      },
+                      {
+                        name: "twitter:card",
+                        content: "summary_large_image",
+                      },
+                    ]
+                  : [
+                      {
+                        name: "twitter:card",
+                        content: "summary",
+                      },
+                    ]
+              )
+              .concat(meta)}
+          />
+        )
+      }}
+    />
+  )
+}
+
+SEO.defaultProps = {
+  meta: [],
+}
+
+SEO.propTypes = {
+  description: PropTypes.string,
+  image: PropTypes.shape({
+    src: PropTypes.string.isRequired(),
+    height: PropTypes.string.isRequired(),
+    width: PropTypes.string.isRequired(),
+  }),
+  meta: PropTypes.array,
+  // highlight-next-line
+  slug: PropTypes.string,
+  title: PropTypes.string.isRequired,
+}
+
+export default SEO
+```
+
+Woo hoo! Lot to digest here, but you've enabled adding an _absolute_ canonical link simply by passing in a `slug` prop and prefixing with `siteUrl`.
+
+To bring it all home, it's high time to begin actually _using_ this extensible SEO component to show all of these moving parts coming together to deliver a great SEO experience.
+
+## Using the SEO component
+
+You created an extensible SEO component. It takes a `title` prop, and then (optionally) `description`, `meta`, `image`, and `pathname` props. Let's wire it all up!
+
+### In a page component
+
+```jsx:title=src/pages/index.js
+import React from "react"
+
+import Layout from "../components/layout"
+import SEO from "../components/seo" // highlight-line
+
+function Index() {
+  return (
+    <Layout>
+      <SEO title="My Amazing Gatsby App" /> {/* highlight-line */}
+      <h1>lol - pretend this is meaningful content</h1>
+    </Layout>
+  )
+}
+
+export default Index
+```
+
+### In a template
+
+Let's pretend you have a Markdown powered blog (see: [this tutorial][gatsby-markdown-blog] for more info). Of course, you want some nice SEO as well as a nifty image for sharing on Twitter, Facebook, and Slack. The following steps are needed:
+
+- Create a Markdown post
+- Add an image, and add it to the Markdown posts frontmatter
+- Query this image with GraphQL
+
+#### Creating the post
+
+```shell
+mkdir -p content/blog/2019-01-04-hello-world-seo
+touch content/blog/2019-01-04-hello-world-seo/index.md
+```
+
+```md:title=content/blog/2019-01-04-hello-world-seo/index.md
+---
+date: 2019-01-04
+featured: images/featured.jpg
+---
+
+Hello World!
+```
+
+#### Adding the image
+
+Feel free to add whatever, or perhaps use this _very pertinent_ image:
+
+![Sample Image](./images/seo.jpg)
```

Comment:
```suggestion
![Sample Image](./images/seo.jpg)
```

If you choose to maintain the sentence immediately preceding this image, I suggest making the alt text more specific. Anyone who can't see the image won't get the joke!